pkg/common/utils.go

```diff
@@ -148,3 +149,60 @@ func CreateNodeID(project, zone, name string) string {
 func CreateZonalVolumeID(project, zone, name string) string {
 	return fmt.Sprintf(volIDZonalFmt, project, zone, name)
 }
+
+// ConvertLabelsStringToMap converts the labels from string to map
+// example: "key1=value1,key2=value2" gets converted into {"key1": "value1", "key2": "value2"}
+func ConvertLabelsStringToMap(labels string) (map[string]string, error) {
+	const labelsDelimiter = ","
+	const labelsKeyValueDelimiter = "="
+
+	labelsMap := make(map[string]string)
+	if labels == "" {
+		return labelsMap, nil
+	}
+
+	regexKey, _ := regexp.Compile("^[a-z][-_a-z0-9]{0,62}$")
```

Comment:
Use \p{Lowercase_Letter} instead of a-z? eg `^\p{Ll}[-_\p{Ll}0-9]{0,62}$`.

That way international characters are dealt with correctly, IIUC: https://www.regular-expressions.info/unicode.html

I tested it in go playground and it seems to work as expected.